### PR TITLE
Adds a new proc that automatically checks for armor and spreads damage across parts, fixes a lot of improper damages, adds support for multi-damage abnormalities

### PR DESCRIPTION
**Pull request by Gboster-0**
## About The Pull Request

Okay so i decided to change literally every abnormality in the game by adding a new proc, it does the following:
- automatically applies armor reduction that abnormalities or people may have to that damage
- Spreads it across the body
- Can be given several types of damage as a list instead of one, applying each one after dividing the damage amount by the amount of damages in that list

Unintentionally (from the start) added support for abnormalities dealing multiple damage types
- If `work_damage_type` is now a list, it will deal all of the damages in that list in accordance to `deal_damage()`, along with giving a random visual following to the damage in the list

## Why It's Good For The Game

Less copy-paste, more gud code, various fixes
Also i think its good for most things to apply damage equally to every body part

## Changelog
:cl:
fix: White womans white hole no longer checks for black armor
fix: Faelanterns root's red damage now respects red armor reduction
fix: Fairy gentleman no longer checks white armor for its black jumping attack
fix: Bottle of Tears no longer checks its own white armor for a black damage attack after saying "Drink Me.~"
fix: Bottle of Tears no longer checks its own white armor for a white attack when you stop drinking its liquid
/:cl:

### CODE REVIEW

**Review by EgorDinamit — DISMISSED** [review]
Overall seems fine to me.
Get a maintainer to merge it.

**Comment by Coxswain-Navigator** [comment]
This is a good PR overall, and I'll address any issues while I am reviewing it. However, I should probably mention that the black damage dealt with a white damage armor check for god of the seasons is intentional - it is meant to damage health + sanity, and the abnormality is expected to deal white damage and thus players are expected to wear white armor.

File: code/modules/mob/living/simple_animal/abnormality/teth/faelantern.dm [hunk 1]
```diff
@@ -236,7 +236,7 @@
 			if(WEST)
 				R.pixel_x -= 16
 		for(var/mob/living/L in T)
-			L.apply_damage(root_damage, RED_DAMAGE, null, spread_damage = TRUE)
+			L.apply_damage(root_damage, RED_DAMAGE, spread_damage = TRUE)
```
Comment (Coxswain-Navigator) [inline code/modules/mob/living/simple_animal/abnormality/teth/faelantern.dm:239]: Is this line meant to use apply_damage and not the new deal_damage proc?

Comment (Gboster-0) [reply code/modules/mob/living/simple_animal/abnormality/teth/faelantern.dm:239]: for documentation = yesn't
deal_damage automatically applies armor reduction, so putting this into that proc raw would result in the armor activelly blocking it, whilst its not the intention
But it is supposed to be changed to BRUTE damage as we use it for penetrating red effects

**Review by Coxswain-Navigator — APPROVED** [review]
I honestly didn't realize this hadn't been merged yet, its good to go.

**Pull request merged**